packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -91,37 +90,66 @@ export type GraphQLRequestClientFactoryConfig = {
   apiKey?: string;
 };
 
+/**
+ * Checks if the provided error is a ClientError.
+ * @param {RetryError} error - The error to be checked.
+ * @returns {boolean} - True if the error is a ClientError, false otherwise.
+ */
+function isClientError(error: ClientError | NodeJS.ErrnoException): error is ClientError {
+  return error instanceof Error && 'response' in error;
+}
+
+/**
+ * Checks if the provided error is a NodeJS.ErrnoException.
+ * @param {RetryError} error - The error to be checked.
+ * @returns {boolean} - True if the error is a NodeJS.ErrnoException, false otherwise.
+ */
+function isNodeErrException(
+  error: ClientError | NodeJS.ErrnoException
+): error is NodeJS.ErrnoException {
+  return error instanceof Error && 'code' in error;
+}
+
 /**
  * Represents a default retry strategy for handling retry attempts in case of specific HTTP status codes.
  * This class implements the RetryStrategy interface and provides methods to determine whether a request
  * should be retried and calculates the delay before the next retry attempt.
  */
 export class DefaultRetryStrategy implements RetryStrategy {
-  private statusCodes: number[];
-  private errorCodes: string[];
+  private clientErrorCodes: number[];
+  private nodeErrorCodes: string[];
   private factor: number;
 
   /**
    * @param {Object} options Configurable options for retry mechanism.
-   * @param {number[]} options.statusCodes HTTP status codes to trigger retries on
-   * @param {string[]} options.errorCodes Node error codes to trigger retries
+   * @param {number[]} options.clientErrorCodes HTTP status codes to trigger retries on
+   * @param {string[]} options.nodeErrorCodes Node error codes to trigger retries
```

Comment:
Isn't the `statusCodes` rename a breaking change?

I thought the original names you had were fine. If we did want to be more specific with the names, I would suggest `httpStatusCodes` and `nodeErrorCodes`